webpack@2.0.7-beta does not satisfy its siblings' peerDependencies requirements

Trying to `npm install` on a freshly downloaded copy, we get:

~~~
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v4.3.0
npm ERR! npm  v2.14.12
npm ERR! code EPEERINVALID

npm ERR! peerinvalid The package webpack@2.0.7-beta does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer extract-text-webpack-plugin@1.0.1 wants webpack@^1.9.11
npm ERR! peerinvalid Peer sass-loader@3.1.2 wants webpack@^1.12.6
npm ERR! peerinvalid Peer istanbul-instrumenter-loader@0.1.3 wants webpack@1.x.x
npm ERR! peerinvalid Peer karma-webpack@1.7.0 wants webpack@^1.4.0
npm ERR! peerinvalid Peer webpack-dev-server@2.0.0-beta wants webpack@>=2.0.3-beta <3

npm ERR! Please include the following file with any support request:
npm ERR!     D:\<some path>\angular2-webpack\npm-debug.log
~~~

Current environment is:

* Windows 10 x64
* NVM 1.1.0
* node 4.3.0
* npm 2.14.12
* tsc 1.0.3.0

No global package installed beside npm and typescript

I see. It seems not. 

Maybe we should wait for a more stable webpack version and that all used plugins catch up with that. ... Having Angular 2 in beta is enough! :smile: 

~~~
npm start

> app@0.0.0 start D:\<some-path>\angular2-webpack
> npm run server


> app@0.0.0 server D:\<some-path>\angular2-webpack
> webpack-dev-server --env dev --inline --progress --port 3000

Config did not export an object.

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "server"
npm ERR! node v4.3.0
npm ERR! npm  v2.14.12
npm ERR! code ELIFECYCLE
npm ERR! app@0.0.0 server: `webpack-dev-server --env dev --inline --progress --port 3000`
npm ERR! Exit status 4294967295
npm ERR!
npm ERR! Failed at the app@0.0.0 server script 'webpack-dev-server --env dev --inline --progress --port 3000'.
npm ERR! This is most likely a problem with the app package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     webpack-dev-server --env dev --inline --progress --port 3000
npm ERR! You can get their info via:
npm ERR!     npm owner ls app
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\<some-path>\angular2-webpack\npm-debug.log

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
npm ERR! node v4.3.0
npm ERR! npm  v2.14.12
npm ERR! code ELIFECYCLE
npm ERR! app@0.0.0 start: `npm run server`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the app@0.0.0 start script 'npm run server'.
npm ERR! This is most likely a problem with the app package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run server
npm ERR! You can get their info via:
npm ERR!     npm owner ls app
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\<some-path>\angular2-webpack\npm-debug.log
~~~

--- a/README.md
+++ b/README.md
@@ -2,7 +2,7 @@
 
 [![Dependency Status](https://david-dm.org/preboot/angular2-webpack/status.svg)](https://david-dm.org/preboot/angular2-webpack#info=dependencies) [![devDependency Status](https://david-dm.org/preboot/angular2-webpack/dev-status.svg)](https://david-dm.org/preboot/angular2-webpack#info=devDependencies)
 
-A complete, yet simple, starter for Angular 2 using webpack 2.
+A complete, yet simple, starter for Angular 2 using Webpack.
 
 This seed repo serves as an Angular 2 starter for anyone looking to get up and running with Angular 2 and TypeScript fast. Using [Webpack](http://webpack.github.io/) for building our files and assisting with boilerplate. We're also using Protractor for our end-to-end story and Karma for our unit tests.
 * Best practices in file and application organization for [Angular 2](https://angular.io/).

--- a/karma.conf.js
+++ b/karma.conf.js
@@ -1,5 +1,5 @@
 var path = require('path');
-var webpackConfig = require('./webpack.config')('test');
+var webpackConfig = require('./webpack.config');
 
 module.exports = function(config) {
     var _config = {

--- a/package.json
+++ b/package.json
@@ -7,8 +7,8 @@
     "clean-install": "npm run clean && npm install",
     "clean-start": "npm run clean && npm start",
     "watch": "webpack --watch --progress --profile",
-    "build": "rimraf dist && webpack --env prod --progress --profile",
-    "server": "webpack-dev-server --env dev --inline --progress --port 3000",
+    "build": "rimraf dist && webpack --progress --profile",
+    "server": "webpack-dev-server --inline --progress --port 3000",
     "webdriver-update": "webdriver-manager update",
     "webdriver-start": "webdriver-manager start",
     "lint": "tsconfig-lint",
@@ -65,7 +65,7 @@
     "typescript": "^1.7.3",
     "typings": "^0.6.8",
     "url-loader": "^0.5.6",
-    "webpack": "^2.0.7-beta",
-    "webpack-dev-server": "^2.0.0-beta"
+    "webpack": "^1.12.13",
+    "webpack-dev-server": "^1.14.1"
   }
 }

--- a/webpack.config.js
+++ b/webpack.config.js
@@ -9,7 +9,13 @@ var HtmlWebpackPlugin = require('html-webpack-plugin');
 var ExtractTextPlugin = require('extract-text-webpack-plugin');
 var CopyWebpackPlugin = require('copy-webpack-plugin');
 
-module.exports = function makeWebpackConfig(ENV) {
+/**
+ * Env
+ * Get npm lifecycle event to identify the environment
+ */
+var ENV = process.env.npm_lifecycle_event;
+
+module.exports = function makeWebpackConfig() {
     /**
      * Config
      * Reference: http://webpack.github.io/docs/configuration.html
@@ -24,14 +30,14 @@ module.exports = function makeWebpackConfig(ENV) {
      */
     if(ENV === 'test') {
         config.devtool = 'inline-source-map';
-    } else if(ENV === 'prod') {
+    } else if(ENV === 'build') {
         config.devtool = 'source-map';
     } else {
         config.devtool = 'eval-source-map';
     }
 
     // add debug messages
-    config.debug = ENV !== 'prod' || ENV !== 'test';
+    config.debug = ENV !== 'build' || ENV !== 'test';
 
     /**
      * Entry
@@ -50,7 +56,7 @@ module.exports = function makeWebpackConfig(ENV) {
         path: root('dist'),
         publicPath: '/',
         filename: 'js/[name].js',
-        chunkFilename: ENV === 'prod' ? '[id].chunk.js?[hash]' : '[id].chunk.js'
+        chunkFilename: ENV === 'build' ? '[id].chunk.js?[hash]' : '[id].chunk.js'
     };
 
     /**
@@ -202,12 +208,12 @@ module.exports = function makeWebpackConfig(ENV) {
             // Extract css files
             // Reference: https://github.com/webpack/extract-text-webpack-plugin
             // Disabled when in test mode or not in build mode
-            new ExtractTextPlugin('css/[name].css', {disable: ENV !== 'prod'})
+            new ExtractTextPlugin('css/[name].css', {disable: ENV !== 'build'})
         );
     }
 
     // Add build specific plugins
-    if(ENV === 'prod') {
+    if(ENV === 'build') {
         config.plugins.push(
             // Reference: http://webpack.github.io/docs/list-of-plugins.html#noerrorsplugin
             // Only emit files when there are no errors
@@ -273,7 +279,7 @@ module.exports = function makeWebpackConfig(ENV) {
     };
 
     return config;
-};
+}();
 
 // Helper functions
 function root(args) {

